Docs: (flat config) improve global ignores documentation

### Environment

Node version: v18.16.1
npm version: v9.5.0
Local ESLint version: v8.55.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 23.1.0

### What parser are you using?

Default (Espree)

### What did you do?

```
module.exports = [
  {
    ignores: ["dist"],
  },
  {
    files: ["src/**/*.js"],
    rules: {
      'no-console': 'error',
    }
  },
]
```


### What did you expect to happen?

Given the folder structure:
```
eslint.config.js
src/not-ignored.js
src/dist/not-ignored.js
```

I would expect that running `eslint src` would lint `src/not-ignored.js` and would skip `src/dist/not-ignored.js`.

### What actually happened?

Both files do get linted

```
$ ESLINT_USE_FLAT_CONFIG=true node_modules/.bin/eslint src

/Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js
  1:1  error  Unexpected console statement  no-console

/Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js
  1:1  error  Unexpected console statement  no-console

✖ 2 problems (2 errors, 0 warnings)
```


### Link to Minimal Reproducible Example

https://github.com/bradzacher/eslint-flat-config-bugs/tree/main/not-ignored-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

probably relevant DEBUG=* output

```
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js (based on ignores: **/node_modules/,.git/) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js (based on ignores: dist) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +1ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  eslint:rules Loading rule 'no-console' (remaining=290) +0ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +5ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js (based on ignores: **/node_modules/,.git/) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js (based on ignores: dist) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
```

It looks like the `src/dist/ignored.js` file gets ignored then unignored by the `files` glob.

This is contrary to what I'd expect [based on the docs that state that](https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores)

> If ignores is used without any other keys in the configuration object, then the patterns act as global ignores.

I wouldn't expect that I can implicilty unignore a file by simply adding it to a `files` - I'd expect that I'd need to explicitly unignore it eg `ignores: ['!src/dist/ignored.js']`

--- a/docs/src/use/configure/ignore.md
+++ b/docs/src/use/configure/ignore.md
@@ -38,7 +38,31 @@ You can also ignore files on the command line using `--ignore-pattern`, such as:
 npx eslint . --ignore-pattern ".config/*"
 ```
 
-## Unignoring Files
+## Ignoring Directories
+
+Ignoring directories works the same way as ignoring files, by placing a pattern in the `ignores` key of a configuration object with no other keys. For example, the following ignores the `.config` directory as a whole (meaning file search will not traverse into it at all):
+
+```js
+// eslint.config.js
+export default [
+    {
+        ignores: [".config/"]
+    }
+];
+```
+
+Unlike `.gitignore`, an ignore pattern like `.config` will only ignore the `.config` directory in the same directory as the configuration file. If you want to recursively ignore all directories named `.config`, you need to use `**/.config/`, as in this example:
+
+```js
+// eslint.config.js
+export default [
+    {
+        ignores: ["**/.config/"]
+    }
+];
+```
+
+## Unignoring Files and Directories
 
 You can also unignore files and directories that are ignored by previous patterns, including the default patterns. For example, this config unignores `node_modules/mylibrary`:
 

--- a/docs/src/use/configure/index.md
+++ b/docs/src/use/configure/index.md
@@ -52,5 +52,6 @@ All of these options give you fine-grained control over how ESLint treats your c
 [**Ignore Files**](ignore)
 
 * [Ignoring Files](./ignore#ignoring-files)
-* [Unignoring Files](./ignore#unignoring-files)
+* [Ignoring Directories](./ignore#ignoring-directories)
+* [Unignoring Files and Directories](./ignore#unignoring-files-and-directories)
 * [Ignored File Warnings](./ignore#ignored-file-warnings)

--- a/docs/src/use/configure/migration-guide.md
+++ b/docs/src/use/configure/migration-guide.md
@@ -524,7 +524,7 @@ config/*
 # ...other ignored files
 ```
 
-`ignorePatterns` example:
+Here are the same patterns represented as `ignorePatterns` in a `.eslintrc.js` file:
 
 ```javascript
 // .eslintrc.js
@@ -534,7 +534,7 @@ module.exports = {
 };
 ```
 
-Here are the same files ignore patterns in flat config:
+The equivalent ignore patterns in flat config look like this:
 
 ```javascript
 export default [
@@ -545,7 +545,11 @@ export default [
 ];
 ```
 
-Also, with flat config, dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add files ignore pattern `"**/.*"`.
+In `.eslintignore`, `temp.js` ignores all files named `temp.js`, whereas in flat config, you need to specify this as `**/temp.js`. The pattern `temp.js` in flat config only ignores a file named `temp.js` in the same directory as the configuration file.
+
+::: important
+In flat config , dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add an ignore pattern of `"**/.*"`.
+:::
 
 ### Linter Options
 

